infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -129,10 +131,37 @@ public LogUnitServer(ServerContext serverContext) {
     /**
      * Service an incoming request for maximum global address the log unit server has written.
      */
-    @ServerHandler(type = CorfuMsgType.TAIL_REQUEST)
-    public void handleTailRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+    @ServerHandler(type = CorfuMsgType.LOG_TAIL_REQUEST)
```

Comment:
I think we should collapse all the query messages into a single generic query request/resp. Adding more message types per query is kind of messy.